### PR TITLE
manim 0.18.0 (new formula)

### PR DESCRIPTION
**Pull request by unitof**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
At last! After [months](https://github.com/Homebrew/homebrew-core/pull/121408) of (sporadic) debugging and half learning a ton about Python packagers and builders and finding [one very sneaky upstream bug](https://github.com/pyglet/pyglet/issues/999): `manim`.

### CODE REVIEW

**Review by SMillerDev — COMMENTED** [review]
Seems from the error it should also depend on cython

**Comment by unitof** [comment]
Yeah, just confused how it built locally. On it!

**Comment by unitof** [comment]
Apparently not cython—it's the same error I got [when working on this previously](https://github.com/Homebrew/homebrew-core/pull/121408#issuecomment-1478287039): [`fatal error: 'cairo.h' file not found`](https://github.com/Homebrew/homebrew-core/actions/runs/8330543426/job/22795676982?pr=166434#step:3:2413).

What's odd is I've only ever seen this error in the Homebrew GitHub runner environment and never locally, even on the same platform (macos 14-arm64 or macos 15-arm64).

If anyone has a clue, would love some help. Otherwise, back to work!

**Comment by unitof** [comment]
Think I may have figured it out: `pkg-config` is require during build time to allow ManimPango to locate `cairo.h`.

@SMillerDev May I request a `CI-no-fail-fast` label? It's now failing on Linux first, which I'll work on next but would like to see if it builds for macOS now.

**Comment by unitof** [comment]
omg vict’ry

squashing!

File: Formula/m/manim.rb [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    depends_on "cmake" => :build
+    depends_on "patchelf" => :build
+  end
+
+  on_arm do
+    depends_on "pango"
+    depends_on "scipy"
+  end
+
+  resource "pyobjc-core" do
```
Comment (SMillerDev) [inline Formula/m/manim.rb:31]: This seems to exist a couple of times in this list

**Review by SMillerDev — APPROVED** [review]

File: Formula/m/manim.rb [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  head "https://github.com/manimCommunity/manim.git", branch: "main"
+
+  depends_on "cython" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "cairo" # for cairo.h
+  depends_on "ffmpeg"
+  depends_on "numpy"
+  depends_on "py3cairo"
+  depends_on "python-setuptools" # for `import pkg_resources`
```
Comment (chenrui333) [inline Formula/m/manim.rb:18]: any issue tracker for dropping setuptools dep?

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@unitof, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Comment by bika-c** [comment]
Hello! Is this package any different than the pip version? It seems like after installing, I can't import it through python. Any help is appreciated!

**Pull request merged**